engine/src/main/scala/cromwell/webservice/CromwellApiHandler.scala

```diff
@@ -89,8 +89,8 @@ class CromwellApiHandler(requestHandlerActor: ActorRef) extends Actor with Workf
       context.parent ! RequestComplete(StatusCodes.OK, WorkflowAbortResponse(id.toString, WorkflowAborted.toString))
     case WorkflowManagerAbortFailure(_, e) =>
       error(e) {
-        //case _: WorkflowNotFoundException => RequestComplete(StatusCodes.NotFound, APIResponse.error(e))
         case _: IllegalStateException => RequestComplete(StatusCodes.Forbidden, APIResponse.error(e))
+        case _: Exception if e.getMessage.endsWith("because no workflow with that ID is in progress") => RequestComplete(StatusCodes.NotFound, APIResponse.error(e))
```

Comment:
Any reason the abort code couldn't generate a `WorkflowNotFoundException` and this code could stay the same?